plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java

```diff
@@ -95,6 +95,35 @@ protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
         }
     }
 
+    /**
+     * Test that we are correctly escaped the wildcard characters while requesting DataBaseMetaData: getSchemas and getTables with params contains pattern.
+     */
+    @Test
+    public void testDatabaseMetadataSearchEscapedWildCardCharacters()
+    {
+        // wildcard characters on schema name
+        assertQuerySucceeds("SHOW TABLES FROM public");
+        assertQueryFails("SHOW TABLES FROM \"publi_\"", ".*Schema 'publi_' does not exist");
+        assertQueryFails("SHOW TABLES FROM \"pu%lic\"", ".*Schema 'pu%lic' does not exist");
+
+        try {
+            assertUpdate("CREATE TABLE abcd (a int, b int, c int) WITH (primary_key = ARRAY['a'])");
+            assertUpdate("CREATE TABLE ab_d (a int, b int, c int) WITH (primary_key = ARRAY['a'])");
+            assertUpdate("CREATE TABLE \"a%d\" (a int, b int, c int) WITH (primary_key = ARRAY['a'])");
+
+            // wildcard characters on table name
+            assertQuerySucceeds("SHOW CREATE TABLE ab_d");
+            assertQuerySucceeds("SHOW CREATE TABLE \"a%d\"");
+            assertQueryFails("SHOW CREATE TABLE \"a%\"", ".*Table 'ignite.public.a%' does not exist");
+            assertQueryFails("SHOW COLUMNS FROM \"a%\"", ".*Table 'ignite.public.a%' does not exist");
+        }
+        finally {
+            assertUpdate("DROP TABLE abcd");
+            assertUpdate("DROP TABLE ab_d");
+            assertUpdate("DROP TABLE \"a%d\"");
```

Comment:
These tables may not exist if the above CREATE TABLE statement failed. Please move CREATE TABLE to before `try`  or use `DROP TABLE IF EXISTS`.